src/js/eventHandling.js

```diff
@@ -0,0 +1,48 @@
+// local storage 
+export const profileData = JSON.parse(localStorage.getItem("uploadProfile")) || { name: "Daniel Haribo" , image: "https://images.unsplash.com/photo-1687360440984-3a0d7cfde903?ixlib=rb-4.0.3&ixid=M3wxMjA3fDF8MHxlZGl0b3JpYWwtZmVlZHw0Nnx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=400&q=60"};
+
+//import
+import { displayUserProfile} from "./domManipulation";
+
+const profileModal = document.getElementById("profile-modal");
+const profileNameInput = document.getElementById("upload-profile-name");
+const ProfilePicInput = document.getElementById("upload-profile-pic-input");
+export const profilePicture = document.getElementById("upload-profile-pic");
+
+
+document.body.addEventListener("click", (e) => {
+    let targetElement = e.target;
+    if (targetElement.closest(".user-profile-detail")) {
+        profileModal.style.display = "block";
+    }
+    if (targetElement.closest("#save-profile-btn")) {
+        if(profileNameInput.value !== "" ){
+            profileData.name = profileNameInput.value;
+        }
+        profileData.image = profilePicture.src;
+        localStorage.setItem('uploadProfile', JSON.stringify(profileData));
+        displayUserProfile(profileData);
+        profileNameInput.value = "";
+
+    }
+    if (targetElement.closest("#cancel-profile-btn")) {
+        profileModal.style.display = "none";
+        profileNameInput.value = "";
+    }
```

Comment:
You can optimize the event delegation and use `case switch`. note: you can't use closest with case switch.
I would like if you did some reading and choose one `if else` or `case switch` and let me know

good starting point - `https://www.oreilly.com/library/view/high-performance-javascript/9781449382308/ch04.html#if-else_versus_switch`